lib/internal/url.js

```diff
@@ -311,8 +311,7 @@ class URL {
   constructor(input, base) {
     // toUSVString is not needed.
     input = `${input}`;
-    if (base !== undefined &&
-        (!base[searchParams] || !base[searchParams][searchParams])) {
+    if (arguments.length === 2) {
```

Comment:
I think we should still allow to explicitly pass `undefined`. I don't know what the spec says, but Chrome accepts it.

Reply:
@targos https://github.com/whatwg/url/blob/master/url.bs#L2592-L2599 implies it doesn't allow it - but I'm all for Chrome compatibility and I'm not sure what the spec reads as. 

What do you think about just removing the `base[searchParams]` and staying compatible with the URL impl at https://github.com/jsdom/whatwg-url/blob/master/lib/URL-impl.js#L12-L17 ?

Reply:
Staying compatible with whatwg-url SGTM. I would actually like us to throw a more precise error saying which URL us invalid but that's out of scope for this PR

Reply:
This is about how we can write "If base is given" and both sound good to me if CI gets green.

Reply:
@TimothyGu What do you think about this?